pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -102,6 +102,11 @@ public class PersistentStickyKeyDispatcherMultipleConsumers extends PersistentDi
 
     @Override
     public synchronized void addConsumer(Consumer consumer) throws BrokerServiceException {
+        if (IS_CLOSED_UPDATER.get(this) == TRUE) {
+            log.warn("[{}] Dispatcher is already closed. Closing consumer {}", name, consumer);
+            consumer.disconnect();
+            return;
+        }
         super.addConsumer(consumer);
```

Comment:
`super.addConsumer(consumer)` already did this check, right? If yes, we should not repeatedly check it.

Reply:
However the `super.addConsumer` return value is `void`. The method in subClass will not fast return if `super.addConsumer` return.  @poorbarcode

Reply:
Is there any other comments :) @poorbarcode

Reply:
@Technoboy- What do you think about this?

Reply:
yes, `super.addConsumer(consumer)` can't fast return.